dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -987,20 +987,48 @@ private void addLocationHeader(HttpHeaders headers, PersistentEntityResourceAsse
     }
 
 
+
+    /**
+     * Execute a PUT request for an entity with id of type Integer;
+     *
+     * curl -X PUT http://<dspace.restUrl>/api/{apiCategory}/{model}/{id}
+     *
+     * Example:
+     * <pre>
+     * {@code
+     *      curl -X PUT http://<dspace.restUrl>/api/core/metadatafield/1
+     * }
+     * </pre>
+     *
+     * @param request     the http request
+     * @param apiCategory the API category e.g. "core"
+     * @param model       the DSpace model e.g. "metadatafield"
+     * @param id          the ID of the target REST object
+     * @param jsonNode    the part of the request body representing the updated rest object
+     * @return the relevant REST resource
+     */
+    @RequestMapping(method = RequestMethod.PUT, value = REGEX_REQUESTMAPPING_IDENTIFIER_AS_DIGIT)
+    public DSpaceResource<RestAddressableModel> put(HttpServletRequest request,
+                                                    @PathVariable String apiCategory, @PathVariable String model,
+                                                    @PathVariable Integer id,
+                                                    @RequestBody JsonNode jsonNode) {
+        return putOneInternal(request, apiCategory, model, id, jsonNode);
```

Comment:
This isn't a specific issue about this PR. I've tried to do a metadatafield or metadataschema PUT requests when the user doesn't have the correct permissions and I get an 404 error. I was expecting to get a 403 Forbidden instead.

Reply:
I can't seem to reproduce the described issue. I've written extra tests to perform the exact behaviour that was described and they succeed: https://github.com/DSpace/DSpace/pull/2291/commits/d450b59899119cd70e32c3c2f788c22c19d0d02a#diff-9e02166d9f889684bb5eba075a991012R373
https://github.com/DSpace/DSpace/pull/2291/commits/d450b59899119cd70e32c3c2f788c22c19d0d02a#diff-d2e1fa933aba2c21b6d604cdbece20e5R239

Reply:
You can test with the Atmire's database - https://www.dropbox.com/s/ovqp394y3vofnwa/entities7-test-db.sql.gz?dl=1
And with a user that doesn't have any record on epersongroup2eperson, like atmirenv+write5a@g****.com.

Trying this PUT http://localhost/spring-rest/api/core/metadataschemas/1
`{"id":1,"prefix":"dc","namespace":"http://dublincore.org/documents/dcmi-terms/"}`

I've got an 404 error, please check out my apache log:

`SERVER - - [11/Feb/2019:15:14:37 +0000] "PUT /spring-rest/api/core/metadataschemas/1 HTTP/1.1" 404 949 "http://localhost/admin/registries/metadata/eperson?pageId=registry-metadatafields-pagination&page=1&pageSize=25&sortDirection=ASC&sortField=id" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36"`

But, like I said, I think this is an Inherited problem, not specific to this PR.

Reply:
@paulo-graca and @Raf-atmire : I've logged this as a bug in this JIRA ticket: https://jira.duraspace.org/browse/DS-4168

As noted in today's meeting, it seems like this bug is unrelated to this specific PR.  So, I'd encourage someone to claim that JIRA ticket and look into it.